### ISSUE TITLE
Icons not rendering

### ISSUE THREAD

**Issue by obeezzy** [lirios/fluid#237]
## Expected Behavior
Icons show up when using the Icon component.

## Current Behavior
No icon is displayed, sometimes being replaced by a diamond (or tilted square).

## Possible Solution
No idea.

## Steps to Reproduce (for bugs only)
Build and run the minimalqmake application.

The results can be seen below. The top right corner of the screen should have an icon, however no icon is displayed.
![no-icon](https://user-images.githubusercontent.com/12936090/39622067-c1f24280-4f45-11e8-8b2e-7ac7c231a31f.png)

In this image, you can see that the icons are represented by diamonds.
![no-icon-rr](https://user-images.githubusercontent.com/12936090/39622302-8316e6aa-4f46-11e8-86d7-a57a3687d8bf.png)

## Environment

Include as many relevant details about the environment you experienced the bug in.

* Version: Qt 5.10.0
* Operating system and version: Ubuntu 16.04
* Device (phone, tablet, laptop, desktop, ...): Laptop
* CPU, RAM, screen resolution: 8 GB of RAM, 1920x1080 resolution

**Comment by obeezzy** [comment]
I think I've discovered the causes of the issues.

For the first issue (the minimal qmake example), it turns out I called the "fetch_icons.sh" script in the wrong directory. Calling it in the right directory fixes that issue.

For the second issue, it seems like the code needed to resolve icons is missing in the "IconThemeImageProvider" class. There is currently no code to handle the display of Material design icons when created with "FluidControls.Icon", even though the [documentation claims it can](https://liri.io/docs/sdk/fluid/develop/qml-fluid-controls-icon.html). I'm working on a fix right now and I will create a pull request once I'm done.

Thanks.
